pfsimulator/parflow_lib/parameter_union.h

```diff
@@ -0,0 +1,233 @@
+/*BHEADER**********************************************************************
+*
+*  Copyright (c) 1995-2024, Lawrence Livermore National Security,
+*  LLC. Produced at the Lawrence Livermore National Laboratory. Written
+*  by the Parflow Team (see the CONTRIBUTORS file)
+*  <parflow@lists.llnl.gov> CODE-OCEC-08-103. All rights reserved.
+*
+*  This file is part of Parflow. For details, see
+*  http://www.llnl.gov/casc/parflow
+*
+*  Please read the COPYRIGHT file or Our Notice and the LICENSE file
+*  for the GNU Lesser General Public License.
+*
+*  This program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License (as published
+*  by the Free Software Foundation) version 2.1 dated February 1999.
+*
+*  This program is distributed in the hope that it will be useful, but
+*  WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms
+*  and conditions of the GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU Lesser General Public
+*  License along with this program; if not, write to the Free Software
+*  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
+*  USA
+**********************************************************************EHEADER*/
+
+/*
+ * ParameterUnion looks for a parameter that might go by
+ * different aliases or assume different types of values.
+ *
+ * It can be used in cases where:
+ * - a physical variable may come under different names
+ * - different physical quantities can provide the same
+ *  information, so when one is defined, all the others
+ *  are readily obtained
+ * - a variable may be defined through different inputs
+ *  (e.g. through a constant value or a filename).
+ *
+ * ParameterUnion will produce an error if none of the
+ * required keys is found. At least one key/alias must be
+ * defined in the input file. In case multiple aliases are
+ * defined in the input file, the first key to be found is
```

Comment:
If we allow aliasing it should be an error to provide multiple aliases for the same value.   Highly likely the user doesn't intend this.